src/modules/flow/iio/nodes.c

```diff
@@ -696,48 +694,56 @@ color_create_channels(struct color_data *mdata, int device_id)
 
 #undef ADD_CHANNEL
 
-    sol_iio_device_start_buffer(mdata->device);
+    sol_iio_device_start_buffer(mdata->iio_base.device);
 
     return true;
 
 error:
     SOL_WRN("Could not create iio/color-sensor node. Failed to open IIO"
         " device %d", device_id);
-    sol_iio_close(mdata->device);
+    sol_iio_close(mdata->iio_base.device);
     return false;
 }
 
 static int
 color_open(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
 {
-    struct color_data *mdata = data;
+    struct iio_color_data *mdata = data;
     const struct sol_flow_node_type_iio_color_sensor_options *opts;
     int device_id;
 
     SOL_FLOW_NODE_OPTIONS_SUB_API_CHECK(options, SOL_FLOW_NODE_TYPE_IIO_COLOR_SENSOR_OPTIONS_API_VERSION,
         -EINVAL);
     opts = (const struct sol_flow_node_type_iio_color_sensor_options *)options;
 
-    mdata->buffer_enabled = opts->buffer_size > -1;
+    mdata->iio_base.buffer_enabled = opts->buffer_size > -1;
 
-    SOL_SET_API_VERSION(mdata->config.api_version = SOL_IIO_CONFIG_API_VERSION; )
+    SOL_SET_API_VERSION(mdata->iio_base.config.api_version = SOL_IIO_CONFIG_API_VERSION; )
 
     if (opts->iio_trigger_name) {
-        mdata->config.trigger_name = strdup(opts->iio_trigger_name);
-        SOL_NULL_CHECK(mdata->config.trigger_name, -ENOMEM);
+        mdata->iio_base.config.trigger_name = strdup(opts->iio_trigger_name);
+        SOL_NULL_CHECK(mdata->iio_base.config.trigger_name, -ENOMEM);
     }
 
-    mdata->config.buffer_size = opts->buffer_size;
-    mdata->config.sampling_frequency = opts->sampling_frequency;
-    if (mdata->buffer_enabled) {
-        mdata->config.sol_iio_reader_cb = color_reader_cb;
-        mdata->config.data = node;
+    mdata->iio_base.config.buffer_size = opts->buffer_size;
+    mdata->iio_base.config.sampling_frequency = opts->sampling_frequency;
+    if (mdata->iio_base.buffer_enabled) {
+        mdata->iio_base.config.sol_iio_reader_cb = color_reader_cb;
+        mdata->iio_base.config.data = node;
     }
-    mdata->use_device_default_scale = opts->use_device_default_scale;
-    mdata->use_device_default_offset = opts->use_device_default_offset;
-    mdata->scale = opts->scale;
-    mdata->offset = opts->offset;
-    mdata->out_range = opts->out_range;
+    mdata->iio_base.use_device_default_scale = opts->use_device_default_scale;
+    mdata->iio_base.use_device_default_offset = opts->use_device_default_offset;
+
+    mdata->scale_red = opts->scale.x;
+    mdata->offset_red = opts->offset.x;
```

Comment:
Why `x` is `red`? There's no correlation between these values. Please don't use direction vector for colours. Use options for this. So you'd had something like:
```
mdata->scale_red = opts->scale_red;
mdata->offset_red = opts->offset_red;
```

Got it?